src/utils.rs

```diff
@@ -1,108 +1,54 @@
+    ($($item:ident),+) => {{
+
+        let mut layouts = Vec::new();
+        $( 
+            let item_layouts =
+                Layout::from_borsh_container(<$item as BorshSchema>::schema_container())
+                    .unwrap();
+
+            layouts.extend(item_layouts);
+        )+
+        layouts
+    }};
 }
 
-/// Generates the TypeScript and borsh layouts of rust data structures found in `.rs`
-/// files.
-pub fn generate_layout_from_file(filepath: impl AsRef<Path>) -> Result<Vec<Layout>, anyhow::Error> {
-    if filepath.as_ref().extension() != Some(OsStr::new("rs")) {
-        return Ok(Vec::new());
-    }
-    let code = fs::read_to_string(filepath)?;
-    let syntax = syn::parse_file(&code)?;
-    let mut layouts = Vec::<Layout>::new();
-    for item in syntax.items {
-        match item {
-            syn::Item::Struct(ref item_struct) => {
-                for attr in &item_struct.attrs {
-                    let attribute_string = attr.tokens.to_string();
-                    if attribute_string.contains(crate::ATTRIBUTE_LABEL) {
-                        layouts.push(Layout::from_tokens(
-                            &item_struct.ident.to_string(),
-                            &mut item_struct.fields.iter(),
-                        )?);
-                    }
-                }
-            }
-            syn::Item::Enum(ref item_enum) => {
-                for attr in &item_enum.attrs {
-                    let attribute_string = attr.tokens.to_string();
-                    if attribute_string.contains(crate::ATTRIBUTE_LABEL) {
-                        let mut enum_layout = Layout {
-                            name: item_enum.ident.to_string(),
-                            kind: Kind::Enum,
-                            fields: Vec::new(),
-                        };
-                        let parent_name = item_enum.ident.to_string();
-                        let mut variant_layouts = item_enum
-                            .variants
-                            .iter()
-                            .map(|variant| {
-                                let name = parent_name.clone() + &variant.ident.to_string();
-                                enum_layout
-                                    .fields
-                                    .push(LayoutField::from_enum_variant(&name)?);
-                                Layout::from_tokens(&name, &mut variant.fields.iter())
-                            })
-                            .collect::<Result<Vec<_>, _>>()?;
-                        layouts.push(enum_layout);
-                        layouts.append(&mut variant_layouts);
-                    }
-                }
-            }
-            _ => {}
-        }
-    }
-    Ok(layouts)
+static LIB_PREABMLE: &str = r#"import { Struct, Enum } from 'ts-borsh-schema';
+import { BinaryReader, BinaryWriter } from "borsh"
+import { PublicKey } from '@velas/web3';
+import BN from "bn.js";
+
+const borshPublicKeyHack = () => {
```

Comment:
unfortunately, this monkeypatch stops to work if used as an external import
https://github.com/dj8yfo/agsol-borsh-schema/pull/2/commits/36ed9af0331ad1980c02565e46bfb4dfd9265297